pkg/plugins/utils_go1.16.go

```diff
@@ -0,0 +1,41 @@
+//go:build !go1.17
+// +build !go1.17
+
```

Comment:
Do you think this file ought to include a note about why it's been copied (and where the code was copied from)?

Reply:
I included a link further down below, but perhaps I should make it stand out more, and move it to a separate comment here?

```go
// fileInfoToDirEntry returns a DirEntry that returns information from info.
 // If info is nil, fileInfoToDirEntry returns nil.
 // taken from https://github.com/golang/go/blob/go1.17/src/io/fs/readdir.go#L49-L77
 func fileInfoToDirEntry(info os.FileInfo) os.DirEntry {
```